src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -779,24 +779,9 @@ ModifyQuerySupported(Query *queryTree, Query *originalQuery, bool multiShardQuer
 		}
 	}
 
-	/*
-	 * We have to allow modify queries with two range table entries, if it is pushdownable.
-	 */
 	if (commandType != CMD_INSERT)
 	{
-		/* We can not get restriction context via master_modify_multiple_shards path */
-		if (plannerRestrictionContext == NULL)
-		{
-			if (queryTableCount != 1)
-			{
-				return DeferredError(ERRCODE_FEATURE_NOT_SUPPORTED,
-									 "cannot run multi shard modify query with master_modify_multiple_shards when the query involves subquery or join",
-									 "Execute the query without using master_modify_multiple_shards()",
-									 NULL);
-			}
-		}
-		/* If it is a multi-shard modify query with multiple tables */
-		else if (multiShardQuery)
+		if (multiShardQuery)
```

Comment:
very minor: maybe `if (commandType != CMD_INSERT && multiShardQuery )` is a little easier to read